RationaleMCP/0031/concept.md

```diff
@@ -0,0 +1,135 @@
+
+## Design Rules 
+
+There are simple general rules for the design of Flat Modelica:
+
+* The syntax is a subset of Modelica syntax with some extensions.
+* Everything which has direct influence on the simulation results in an ideal mathematical sense has a special syntactical construct in the language.
+* Information which has no direct influence to the result are handeled uniformly by one syntactical construct called annotation. For example, this applies to
+   * prefixes
+   * attributes of variables 
+   * information lost in preceding transformation steps,
+   * information to control further transformation steps,
+   * information to support the numerical solution of the equation system
+   * source code location information, 
+   * other data for special use cases, e.g. packaging into a FMU, inputs and outputs of the whole model.
+* Description strings (also called syntactic comments) are an exception. These are optionally copied as such.
+* An external tool does not need any knowledge of special Modelica semantics, e.g. handling start attributes, behavior of relations and built-in operators for events,etc..
+
+## Details
+
+* All higher level syntactical and structural contructs of Modelica are removed, such as:
+   * imports
+   * inheritance
+   * redeclarations
+   * nested class definitions (There is no class definition inside another class definition)
+   * packages, [expandable] connectors, models (only one model remains), blocks, operator definitions
+   * functional inputs of functions
+   * conditional declarations
+   * modifications (therefore no merging of them)
+   * connect statements
+   * loops inside equation sections
+   * `if`/`then`/`else` in equation sections (atleast with different sized branches)
+   * `when` clauses
+   * stream operators: `actualStream(...)`, `inStream(...)`
+   * all prefixes, except `input` and `output`
+* __TODO__ synchronous Modelica extensions, including state machines, possibly introduction of partitions
+
+### Names
+
+* Names are defined as the `IDENT` token in Modelica
+* The hierarchy of the Modelica model is encoded into the names of the variables and types by separation of the levels by a dot.
+* Names from Modelica are generally enclosed in single quotation marks, whether or not they contain a dot.  
+* Generated helper variables do not need single quotes. 
+* __TODO__: array components, combining names for helper variables
+
+### Types
+
+* Following types exist:
+   * predefined types: `Real`, `Integer`, `Boolean`, `String`, `ExternalObject`,
+   * record types
+   * functions
+   * simple types.
+   
+* Properties of record definitions are:
+   * The components are defined as variable definitions. Declaration equations are not allowed.
+   * There are no equations or algorithms.
+   * record definitions can not nest.
```

Comment:
I think it should be clarified that this doesn't rule out having record members of record or array type.

Reply:
Can be made explicit, the variable definition can be of record type and can have a dimensions.